erpnext/stock/doctype/stock_ledger_entry/stock_ledger_entry.py

```diff
@@ -117,6 +118,11 @@ def validate_and_set_fiscal_year(self):
 		if not self.fiscal_year:
 			self.fiscal_year = get_fiscal_year(self.posting_date, company=self.company)[0]
 
+	def validate_leaf_warehouse(self):
```

Comment:
Same function exists in bin, put the function in stock.utils and call from both file. Also, you have introduced a new term "leaf", which is nowhere used, may be you can rename it to "block_transactions_against_group_warehouse" and "is_leaf_warehouse" to "is_group_warehouse".